db-client-java/src/main/java/com/eventstore/dbclient/EventStoreDBPersistentSubscriptionsClient.java

```diff
@@ -314,125 +234,10 @@ public CompletableFuture replayParkedMessagesToAll(String groupName) throws Exec
-        info.setAverageItemsPerSecond(node.get("averageItemsPerSecond").asDouble());
-        info.setTotalItemsProcessed(node.get("totalItemsProcessed").asLong());
-        info.setLastKnownEventNumber(node.get("lastKnownEventNumber").asLong());
-
-        if (node.get("connectionCount") != null) {
-            info.setConnectionCount(node.get("connectionCount").asLong());
-        } else {
-            info.setConnectionCount(0);
-        }
-
-        info.setTotalInFlightMessages(node.get("totalInFlightMessages").asLong());
-        info.setConfig(parseConfig(node.get("config")));
-
-        return info;
-    }
-
-    private static PersistentSubscriptionConfig parseConfig(JsonNode node) {
-        PersistentSubscriptionConfig config = null;
-
-        if (node != null) {
-            config = new PersistentSubscriptionConfig();
-            config.setResolveLinktos(node.get("resolveLinktos").asBoolean());
-            config.setStartFrom(node.get("startFrom").asLong());
-            config.setMessageTimeoutMilliseconds(node.get("messageTimeoutMilliseconds").asLong());
-            config.setExtraStatistics(node.get("extraStatistics").asBoolean());
-            config.setMaxRetryCount(node.get("maxRetryCount").asLong());
-            config.setLiveBufferSize(node.get("liveBufferSize").asLong());
-            config.setBufferSize(node.get("bufferSize").asLong());
-            config.setReadBatchSize(node.get("readBatchSize").asLong());
-            config.setPreferRoundRobin(node.get("preferRoundRobin").asBoolean());
-            config.setCheckPointAfterMilliseconds(node.get("checkPointAfterMilliseconds").asLong());
-            config.setMinCheckPointCount(node.get("minCheckPointCount").asLong());
-            config.setMaxCheckPointCount(node.get("maxCheckPointCount").asLong());
-            config.setMaxSubscriberCount(node.get("maxSubscriberCount").asLong());
-            config.setNamedConsumerStrategy(ConsumerStrategy.valueOf(node.get("namedConsumerStrategy").asText()));
-        }
-
-        return config;
+        return restartSubsystem(RestartPersistentSubscriptionSubsystemOptions.get());
     }
 
-    private static void throwOnError(int code) {
-        switch (code) {
-            case 401:
-                throw new RuntimeException("Access denied");
-            case 404:
-                throw new ResourceNotFoundException();
-            default:
-                if (code >= 200 && code < 300)
-                    return;
-
-                throw new RuntimeException("Unexpected exception, code: " + code);
-        }
-    }
-
-    private static String urlEncode(String value) {
-        try {
-            return URLEncoder.encode(value, StandardCharsets.UTF_8.toString());
-        } catch (UnsupportedEncodingException e) {
-            throw new RuntimeException(e);
-        }
+    public CompletableFuture restartSubsystem(RestartPersistentSubscriptionSubsystemOptions options) {

```

Comment:
**SUGGESTION:** It'd be good to add at least basic test checking if this method is not failing.

Reply:
The test was already there: https://github.com/EventStore/EventStoreDB-Client-Java/blob/51d927224c7b908a1958912d04ec99d00d910fa7/db-client-java/src/test/java/com/eventstore/dbclient/PersistentSubscriptionManagementTests.java#L301